Packs/Tenable_io/Integrations/Tenable_io/Tenable_io.py

```diff
@@ -1114,33 +1248,234 @@ def export_vulnerabilities_command(args: Dict[str, Any]) -> PollResult:
+        filterQuality:  =,!=
+        filterValue:    test,test2
+
+        returns:
+        {
+            'filter.0.filter': 'name',
+            'filter.0.quality': '=',
+            'filter.0.value': 'test',
+            'filter.1.filter': 'description',
+            'filter.1.quality': '!=',
+            'filter.1.value': 'test2',
+        }
+    '''
+    filters: dict[str, str] = {}
+    filter_lists = map(argToList, map(args.get, ('filterName', 'filterQuality', 'filterValue')))
+    for i, (name, quality, value) in enumerate(zip(*filter_lists)):
+        filters |= {
+            f'filter.{i}.filter': name,
+            f'filter.{i}.quality': quality,
+            f'filter.{i}.value': value,
+        }
+
+    return filters
+
+
+def export_scan_body(args: dict) -> dict:
+
+    if chapters := args.get('chapters'):
+        chapters = ';'.join(argToList(chapters))
+    elif args['format'] in ('PDF', 'HTML'):
+        raise DemistoException('The "chapters" field must be provided for PDF or HTML formats.')
+
+    body = {
+        'format': args['format'].lower(),
+        'chapters': chapters,
+        'filter.search_type': args['filterSearchType'].lower(),
+        'asset_id': args.get('assetId'),
+    } | build_filters(args)
+    return body
+
+
+def initiate_export_scan(args: dict, client: Client) -> str:
+    return client.initiate_export_scan(
+        args['scanId'],
+        params=sub_dict(args, 'historyId', 'historyUuid'),
+        body=export_scan_body(args)
+    ).get('file', '')
+
+
+@polling_function(
+    'tenable-io-export-scan',
+    poll_message='Preparing scan report:',
+    interval=15,
+    requires_polling_arg=False)
+def export_scan_command(args: dict[str, Any], client: Client) -> PollResult:
+    '''
+    Calls three endpoints. The first (called with initiate_export_scan) initiates an export and returns a file ID.
+    The second (called with client.check_export_scan_status) checks the status of the export and the function polls
+    until the status is 'ready'. The third endpoint is then called (with client.download_export_scan) which downloads
+    the file and returns a dict with it's contents (using fileResult).
+    '''
+
+    scan_id = args['scanId']
+    file_id = (
+        args.get('fileId')
+        or initiate_export_scan(args, client))
+
+    match client.check_export_scan_status(scan_id, file_id).get('status'):
+        case 'ready':
+            return PollResult(
+                client.download_export_scan(
+                    scan_id, file_id, args['format']),
+                continue_to_poll=False)
+
+        case 'loading':
+            return PollResult(
+                None,
+                continue_to_poll=True,
+                args_for_next_run={
+                    'fileId': file_id,
+                    'scanId': scan_id,
+                    'format': args['format'],  # not necessary but avoids confusion
+                })
+
+        case error:
```

Comment:
why are there no quotes around error?

Reply:
it also takes into account an unrecognised status

Reply:
yes